compiler/modules/types/mapping.bal

```diff
@@ -32,6 +32,13 @@ public class MappingDefinition {
     }
 
     public function define(Env env, Field[] fields, SemType rest) returns SemType {
+            Field[] cellFields = from Field f in fields select [f[0], cellContaining(env, f[1], CELL_MUT_LIMITED)];
+            SemType restCell = cellContaining(env, rest, CELL_MUT_LIMITED);
+            return self.defineWithoutWrapper(env, cellFields, restCell);
+    }
+
+    // Calling directly will not have the members wrapped in cells
+    public function defineWithoutWrapper(Env env, Field[] fields, SemType rest) returns SemType {
```

Comment:
This should be called `define` and the function that wraps should be called something else. This is for consistency for the accessing functions, where the unwrapping function uses deref.

Reply:
Please check with the latest changes.